client/components/data/query-email-forwards/index.jsx

```diff
@@ -0,0 +1,55 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import isRequestingEmailForwards from 'state/selectors/is-requesting-email-forwards';
+import { getEmailForwards } from 'state/email-forwarding/actions';
+
+class QueryEmailForwards extends PureComponent {
+	static propTypes = {
+		domainName: PropTypes.string.isRequired,
+		getEmailForwards: PropTypes.func.isRequired,
+		requestingEmailForwards: PropTypes.bool.isRequired,
+	};
+
+	componentDidMount() {
```

Comment:
you'll also want to add a check in `componentDidUpdate` to see if the domainName changed and re-request.